feature-libs/user/profile/components/reset-password/reset-password-component.service.spec.ts

```diff
@@ -82,166 +85,251 @@ describe('ResetPasswordComponentService', () => {
-          error.details = [{ message: 'error message' }];
-          spyOn(userPasswordService, 'reset').and.returnValue(
-            throwError(() => error)
+        it('should reset password', () => {
+          spyOn(userPasswordService, 'reset').and.callThrough();
+          service.resetPassword(resetToken);
+          expect(userPasswordService.reset).toHaveBeenCalledWith(
+            resetToken,
+            'QwePas123!'
           );
+        });
+
+        it('should show message', () => {
           service.resetPassword(resetToken);
           expect(globalMessageService.add).toHaveBeenCalledWith(
-            { raw: 'error message' },
-            GlobalMessageType.MSG_TYPE_ERROR
+            { key: 'forgottenPassword.passwordResetSuccess' },
+            GlobalMessageType.MSG_TYPE_CONFIRMATION
           );
         });
 
-        it('should not show error message', () => {
-          spyOn(userPasswordService, 'reset').and.returnValue(
-            throwError(() => null)
-          );
+        it('should reroute to the login page', () => {
           service.resetPassword(resetToken);
-          expect(globalMessageService.add).not.toHaveBeenCalled();
+          expect(routingService.go).toHaveBeenCalledWith({ cxRoute: 'login' });
         });
 
-        it('should not show error message', () => {
-          spyOn(userPasswordService, 'reset').and.returnValue(
-            throwError(() => ({}))
-          );
+        it('should reset form', () => {
+          spyOn(service.form, 'reset').and.callThrough();
           service.resetPassword(resetToken);
-          expect(globalMessageService.add).not.toHaveBeenCalled();
+          expect(service.form.reset).toHaveBeenCalled();
         });
       });
-    });
 
-    it('should not reset invalid form', () => {
-      spyOn(userPasswordService, 'reset').and.returnValue(
-        throwError(() => ({}))
-      );
-      passwordConfirm.setValue('Diff123!');
-      service.resetPassword(resetToken);
-      expect(userPasswordService.reset).not.toHaveBeenCalled();
-      expect(globalMessageService.add).not.toHaveBeenCalled();
-      expect(routingService.go).not.toHaveBeenCalled();
+      describe('error', () => {
+        describe('valid form', () => {
+          beforeEach(() => {
+            password.setValue('QwePas123!');
+            passwordConfirm.setValue('QwePas123!');
+          });
+
+          it('should show error message', () => {
+            const error = new HttpErrorModel();
+            error.details = [{ message: 'error message' }];
+            spyOn(userPasswordService, 'reset').and.returnValue(
+              throwError(() => error)
+            );
+            service.resetPassword(resetToken);
+            expect(globalMessageService.add).toHaveBeenCalledWith(
+              { raw: 'error message' },
+              GlobalMessageType.MSG_TYPE_ERROR
+            );
+          });
+
+          it('should not show error message', () => {
```

Comment:
duplicated name `should not show error message`

Reply:
fixed